epic_connection.py

```
@@ -0,0 +1,238 @@
+                        logger.error(
+                            "Length of data received doesn't match size "
+                            "information received ahead of time: "
+                            f"{len(data_msg.encode())} != {size_data}"
+                        )
+                    else:
+                        try:
+                            json_data = json.loads(data_msg)
+                        except TypeError:
+                            logger.error(
+                                f"Received {data_msg} but not in JSON format"
+                            )
```

Ruff (0.14.7)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)